coderd/inboxnotifications.go

```diff
@@ -0,0 +1,376 @@
+		for _, template := range splitTemplates {
+			id, err := uuid.Parse(template)
+			if err != nil {
+				logger.Error(ctx, "unable to parse template id", slog.Error(err))
+				return nil, nil, "", xerrors.New("unable to parse template id")
+			}
+			templates = append(templates, id)
+		}
+	}
+
+	if readStatusParam != "" {
+		readOptions := []string{
+			string(database.InboxNotificationReadStatusRead),
+			string(database.InboxNotificationReadStatusUnread),
+			string(database.InboxNotificationReadStatusAll),
+		}
+
+		if !slices.Contains(readOptions, readStatusParam) {
+			logger.Error(ctx, "unable to parse read status")
+			return nil, nil, "", xerrors.New("unable to parse read status")
+		}
+	}
+
+	return targets, templates, readStatusParam, nil
+}
+
+// convertInboxNotificationResponse works as a util function to transform a database.InboxNotification to codersdk.InboxNotification
+func convertInboxNotificationResponse(ctx context.Context, logger slog.Logger, notif database.InboxNotification) codersdk.InboxNotification {
+	return codersdk.InboxNotification{
+		ID:         notif.ID,
+		UserID:     notif.UserID,
+		TemplateID: notif.TemplateID,
+		Targets:    notif.Targets,
+		Title:      notif.Title,
+		Content:    notif.Content,
+		Icon:       notif.Icon,
+		Actions: func() []codersdk.InboxNotificationAction {
+			var actionsList []codersdk.InboxNotificationAction
+			err := json.Unmarshal([]byte(notif.Actions), &actionsList)
+			if err != nil {
+				logger.Error(ctx, "unmarshal inbox notification actions", slog.Error(err))
+			}
+			return actionsList
+		}(),
+		ReadAt: func() *time.Time {
+			if !notif.ReadAt.Valid {
+				return nil
+			}
+			return &notif.ReadAt.Time
+		}(),
+		CreatedAt: notif.CreatedAt,
+	}
+}
+
+// watchInboxNotifications watches for new inbox notifications and sends them to the client.
+// The client can specify a list of target IDs to filter the notifications.
+// @Summary Watch for new inbox notifications
+// @ID watch-for-new-inbox-notifications
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Notifications
+// @Param targets query string false "Comma-separated list of target IDs to filter notifications"
+// @Param templates query string false "Comma-separated list of template IDs to filter notifications"
+// @Param read_status query string false "Filter notifications by read status. Possible values: read, unread, all"
+// @Success 200 {object} codersdk.GetInboxNotificationResponse
+// @Router /notifications/inbox/watch [get]
+func (api *API) watchInboxNotifications(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var (
+		apikey          = httpmw.APIKey(r)
+		targetsParam    = r.URL.Query().Get("targets")
+		templatesParam  = r.URL.Query().Get("templates")
+		readStatusParam = r.URL.Query().Get("read_status")
+	)
+
+	targets, templates, readStatusParam, err := convertInboxNotificationParameters(ctx, api.Logger, targetsParam, templatesParam, readStatusParam)
```

Comment:
Did you try to use https://github.com/go-playground/validator with a request type like we do elsewhere?

See https://github.com/coder/coder/blob/8c0350e20cbf84168592a6715079bdbc22aa4e41/codersdk/organizations.go#L211-L223 (definition) and https://github.com/coder/coder/blob/f651ab937b25eab04833e955f32412b1f83893e7/coderd/workspaces.go#L361-L364 (usage).